### DISCUSSION TITLE
Web midi.js implementing a bass coupler

### DISCUSSION THREAD

**Discussion by jsCoder07** [djipco/webmidi#356] [Q&A]
I am trying to implement a bass and melody coupler. They usually read what the lowest or highest notes on a chord are and then play the last bass note an octave lower. The second detects the last note of the chord and plays that note with a more prominent voice.

I am not sure how to read the last note or the first note of a chord from the midi message as they come in.

Please I'd appreciate your help

**Comment by tobyhede** [comment]
You probably need to intercept `sendNoteOn` or `playNote`.
https://github.com/djipco/webmidi/blob/master/src/OutputChannel.js#L929

Both accept an array of Notes - if a chord is being played I would expect the notes to be in an array. Worth checking to confirm that though.

Once you have the array of notes, you can sort by `number`. 
https://github.com/djipco/webmidi/blob/master/src/Note.js#L271

Once you have the number, there is an `offsetNumber` helper in Utilities for calculating an offset by octave and/or semitone.  
https://github.com/djipco/webmidi/blob/master/src/Utilities.js#L472 

Create a new note with the right octave offset, append to the array and everything continues from there.
